app/models/telephone.rb

```diff
@@ -1,7 +1,8 @@
 class Telephone < ApplicationRecord
   belongs_to :contact
-  validates :number, presence: true, uniqueness: true
-  VALID_TYPES = ['home', 'work', 'other']
-  validates :type, presence: true, inclusion: { in: VALID_TYPES}
-  validates :main, presence: true
+  validates :number, presence: true, uniqueness: {scope: :contact_id}
+  enum :valid_types,  [:home, :work, :other]
+  validates :kind, presence: true, inclusion: { in: valid_types }
```

Comment:
Você pode fazer algo assim aqui:
```suggestion
  enum kind: { home: 'home', work: 'work', other: 'other' }
```
No caso, o uso do enum já vai fazer a validação pra ti dos valores no campo, e ainda vai criar métodos auxiliares úteis com base nos valores, como por exemplo `telephone.home?` que vai checar se aquele telefone é do `kind` `home`.

No exemplo que eu dei ali em cima é usado um hash (chave/valor) pra definir o enum, você pode passar um `array` de símbolos direto, mas aí a gente cai em problemas de ordem dos elementos, o que pode ser bem ruim, como é dito nesse [cop do rubocop](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsenumhash) e [aqui](https://rails.rubystyle.guide/#enums) por exemplo.